src/foam/nanos/boot/Boot.java

```diff
@@ -13,23 +13,25 @@
 import foam.dao.java.JDAO;
 import foam.dao.ProxyDAO;
 import foam.nanos.auth.Group;
-import foam.nanos.auth.Permission;
 import foam.nanos.auth.User;
 import foam.nanos.logger.Logger;
 import foam.nanos.logger.ProxyLogger;
 import foam.nanos.logger.StdoutLogger;
 import foam.nanos.script.Script;
 import foam.nanos.session.Session;
-import java.util.ArrayList;
-import java.util.List;
+
+import java.util.*;
+
 import static foam.mlang.MLang.EQ;
 
 public class Boot {
   // Context key used to store the top-level root context in the context.
   public final static String ROOT = "_ROOT_";
+  private final static Set<String> RELOADED_SERVICES = new HashSet<>();
```

Comment:
Why is this static? This would prevent me from booting two NANOS servers in the same VM. Can't it just be an instance variable?

Reply:
It needs a way to tell if a service is reloaded so that the already loaded web agents and nano services can be reloaded - https://github.com/foam-framework/foam2/pull/2423/commits/9649657c7c8f74325a77ce9d6ac498133088a18d.

It is not an instance variable because we do not pass the instance of Boot to NanoRouter/NanoServiceRouter.

> Why is this static? This would prevent me from booting two NANOS servers in the same VM.
I'm open to change. What are the other alternative you have in mind?

Reply:
Changed to `reloadedServices_` instance variable and store it in the context to be accessible in NanoRouter and NanoServiceRouter.